odoo/tools/misc.py

```diff
@@ -1530,3 +1532,12 @@ def traverse_containers(val, type_):
     elif isinstance(val, collections.abc.Sequence):
         for v in val:
             yield from traverse_containers(v, type_)
+
+
+def sign_data(env, data):
```

Comment:
I was not convinced of the interrest of having an helper method for this but then I saw we have some bad uses in some modules I won't name... If we are going to add one, we should start using it everywhere.
Naming semantic: cryptographic signature is for integrity. hmac is for authentication of message (and integrity as bonus).
Still semantic: never call a variable `data`. `message` is maybe more appropriate.